tests/gold.c

```diff
@@ -0,0 +1,10 @@
+// Test gold support
```

Comment:
Some extra thoughts: we should check (at configure time) that the gold linker is available, and `REQUIRE` ir.